src/gsNurbs/gsBSpline.h

```diff
@@ -22,6 +22,7 @@
 
 #include <gsUtils/gsPointGrid.h>
 
+#include <gsNurbs/gsCurveCurveIntersection.h>
```

Comment:
Please add here a forward declaration
template<class T> gsCurveCurveIntersection;

and move the 
#include <gsNurbs/gsCurveCurveIntersection.h>
to the hpp file

Reply:
A forward declaration for `gsCurveIntersectResult` is added in `gsBSpline.h`, and `#include <gsNurbs/gsCurveCurveIntersection.h>` is added into the `gsBSpline.hpp` file. However, it does not work if I remove the header file `#include <gsNurbs/gsCurveCurveIntersection.h>` from `gsBSpline.h`. So I leave it there. Do you have any suggestions?